### PR TITLE
Deploy hyperkube as docker container or binary

### PR DESCRIPTION
**Pull request by spiffxp**
tl;dr set hyperkube_deployment_mode to 'docker' if you want to run the same way as before

- Add kubernetes_binaries_uri
- Add install-hyperkube.service
- Rename kubectl.service to install-kubectl.service
- Sort ansible inventory variables
- Rename hyperkube-*.service to hyperkube-docker-*.service
- Add hyperkube-binary-*.service
- Rename hyperkube_config_master.json to kubernetes-master-pod.json
- Explicitly start cadvisor service on etcd and master instances
- Extract cadvisor task
- Move master task to master-docker
- Add master-binary task
- Add hyperkube_deployment_mode to ansible (default 'binary')
- Move node task to node-docker
- Add node-binary task
- Rename kraken task to kraken-services
- Make wait4node.service depend on network-online.target
- Drop kubernetes_version as a variable

### CODE REVIEW

File: terraform/local/variables.tf [hunk 1]
```diff
@@ -61,9 +61,17 @@ variable "dockercfg_base64" {
...
 variable "hyperkube_image" {
   default = "gcr.io/google_containers/hyperkube:v1.0.3"
-  description = "docker image to launch hyperkube"
+  description = "image to use when running with hyperkube_deploy_mode 'docker'"
+}
+variable "kubernetes_binaries_uri" {
+  default = "https://storage.googleapis.com/kubernetes-release/release/v1.0.3/bin/linux/amd64"
+  description = "url to fetch hyperkube, kubectl binaries from in hyperkube_deploy_mode 'binary'"
 }
 variable "kubernetes_version" {
```
Comment (maratoid) [inline terraform/local/variables.tf]: Is this still used?

File: ansible/roles/kubernetes/defaults/main.yaml [hunk 2]
```diff
@@ -4,6 +4,8 @@ dns_domain: kubernetes.local
 dns_ip: 10.100.0.10
 dockercfg_base64:
 hyperkube_image: gcr.io/google_containers/hyperkube:v1.0.3
+hyperkube_deployment_mode: binary
+kubernetes_binaries_uri: https://storage.googleapis.com/kubernetes-release/release/v1.0.3/bin/linux/amd64
 kubernetes_version: 1.0.3
```
Comment (maratoid) [inline ansible/roles/kubernetes/defaults/main.yaml]: Is this still used?
